app/src/main/java/com/github/polybooks/ListSaleActivity.kt

```diff
@@ -0,0 +1,58 @@
+package com.github.polybooks
+
+import android.content.Intent
+import androidx.recyclerview.widget.RecyclerView
+import com.github.polybooks.core.Sale
+import com.github.polybooks.core.SaleState
+import com.github.polybooks.core.database.SalesAdapter
+import com.github.polybooks.core.database.interfaces.Query
+import com.github.polybooks.core.database.interfaces.SaleSettings
+import com.github.polybooks.utils.setupNavbar
+import com.google.android.material.bottomnavigation.BottomNavigationView
+
+class ListSaleActivity: ListActivity<Sale>() {
+
+    override fun adapter(list: List<Sale>): RecyclerView.Adapter<*> {
+        return SalesAdapter(list)
+    }
+
+    override fun setNavBar() {
+        val navBarListener : BottomNavigationView.OnNavigationItemSelectedListener =
+            BottomNavigationView.OnNavigationItemSelectedListener{ item ->
+                when(item.itemId){
+                    R.id.home ->{
+                        startActivity(Intent(this, MainActivity::class.java))
+                        true
+                    }
+                    R.id.books ->{
+                        startActivity(Intent(this, ListBooksActivity::class.java))
+                        true
+                    }
+                    R.id.user_profile ->{
+                        // TODO: user sales
+                        false
+                    }
+                    else -> true
+                }
+            }
+        setupNavbar(findViewById(R.id.bottom_navigation), this, R.id.sales, navBarListener)
+    }
+
+    override fun getQuery(): Query<Sale> {
+        return intent.getSerializableExtra(EXTRA_SALE_QUERY_SETTINGS)
+            ?.let {
+                salesDB.querySales()
+                    .fromSettings(intent.getSerializableExtra(EXTRA_SALE_QUERY_SETTINGS) as SaleSettings)
```

Comment:
https://stackoverflow.com/questions/60485368/kotlin-difference-between-it-and-this-keyword

I think you can use "it" keyword instead of rewriting intent.getSerializableExtra

Reply:
Didn't know about it. Neat!